### PR TITLE
Cart coords fix

### PR DESCRIPTION
**Pull request by samo-lin**
Fix the function name to comply with the Julia convention of the usage of '!'.

### CODE REVIEW

**Comment by simonbyrne** [comment]
Can you add a deprecation to deprecated.jl?

File: src/deprecated.jl [hunk 1]
```diff
@@ -9,7 +9,7 @@ const FINALIZE_ATEXIT = Ref(true)
 @deprecate Info_get(info::Info,key::AbstractString) info[Symbol(key)] false
 @deprecate Info_delete(info::Info,key::AbstractString) delete!(info, Symbol(key)) false
 @deprecate Info_free(info::Info) free(info) false
-
+@deprecate Cart_coords!(comm::Comm, maxdims::Integer) Cart_coords(comm, maxdims) false
```
Comment (samo-lin) [inline src/deprecated.jl]: @simonbyrne : is this how it should be (coords allocated in the function and then returned)?

Comment (simonbyrne) [reply src/deprecated.jl]: Yes, that looks good, thanks.

**Comment by simonbyrne** [comment]
Thanks!

**Pull request merged**